src/types/mod.rs

```diff
@@ -169,9 +169,11 @@ impl Flag<'static> {
         }
     }
 
-    /// Helper function to transform a [`Vec`] of flag strings into a [`Vec`] of [`Flag`].
-    pub fn from_vec<S: ToString>(v: &[S]) -> Vec<Self> {
-        v.iter().map(|s| Self::from(s.to_string())).collect()
+    /// Helper function to transform Strings into owned Flags
+    pub fn from_strs<S: ToString>(
```

Comment:
I was specifically thinking we'd want `where Flag<'a>: From<S>` and place the `impl` on `Flag<'a>` (rather than `Flag<'static>`) so that this would automatically do the right thing both for `String` and for `&str`. The way it is now calling `from_strs` on an iterator over `String` will clone each `String` through a call to `to_string`. I _think_
```rust
impl Flag<'a> {
    pub fn from_strs<S>(v: impl IntoIterator<Item = S>) -> impl Iterator<Item = Self> where Flag<'a>: From<S> {
      v.into_iter().map(|s| Flag::from(s))
    }
}
```
should do it?

Reply:
Hmm. I did try with the version you had pasted in there, but it is hard to actually use because everywhere we want to use this we need a copying version because the destination is `'static`, and we usually are being given a `Cow` here that has non static lifetime.

We need an owning version in `UnsolicitedResponse` because it has to send it down a `channel`. So that's that.

For the uses in `parse.rs`, we are *already* copying ([here](https://github.com/jonhoo/rust-imap/blob/4108917c58c4fe5083418b00b6a337e8d3b04ece/src/parse.rs#L113), [here](https://github.com/jonhoo/rust-imap/blob/4108917c58c4fe5083418b00b6a337e8d3b04ece/src/parse.rs#L275) and [here](https://github.com/jonhoo/rust-imap/blob/4108917c58c4fe5083418b00b6a337e8d3b04ece/src/parse.rs#L275)). And we're already copying because we actually need to. Or, at least I could not make the compiler happy, since it complained about not being able to resolve `'a` from the source and `'static` from the dest. I assume this is also why this code is copying already.

So since everywhere we were using this we ended up copying, and we were being given a `Cow`, I didn't end up using the non-copying version, so I removed it as dead code.

Does this make sense? It is possible I just couldn't make it work and am missing something obvious.

Reply:
Oh, I see what you mean. That's unfortunate, but does make sense — I don't think you're missing anything. I'll stop pushing my crazy generics on you :sweat_smile: